backend/models/user.js

```diff
@@ -15,6 +15,38 @@ const mongoose = require("mongoose");
 const crypto = require("crypto");
 const uniqueValidator = require("mongoose-unique-validator");
 
+// Define the UserRank schema
+const userRankSchema = new mongoose.Schema({
+  userRankID: { type: String, required: true, unique: true },
+  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
+  rank: { type: mongoose.Schema.Types.ObjectId, ref: 'Rank', required: true },
+  app: { type: mongoose.Schema.Types.ObjectId, ref: 'App', required: true },
+  dateAchieved: { type: Date, required: true },
+  isActive: { type: Boolean, default: false },
+});
+
+// Define the UserActivity schema
+const userActivitySchema = new mongoose.Schema({
```

Comment:
You'd better create new file userActivity.js for userActivitySchema definition.
We will have many schemas and we couldn't put all of them in one file.
It is not a good approach for project maintainability.